cmake/common/targets.cmake

```diff
@@ -51,6 +51,7 @@ function(silkworm_library TARGET)
   list(FILTER TEST_SRC INCLUDE REGEX "${TEST_REGEX}")
 
   list(FILTER SRC EXCLUDE REGEX "${TEST_REGEX}")
+  list(FILTER SRC EXCLUDE REGEX "cmd/")
```

Comment:
as discussed, cmd lib sources have to be excluded from the parent lib, but included into the cmd lib itself, and also it requires to define the parent lib before the cmd lib, so it is not possible to call `add_subdirectory(cmd)` before declaring `silkworm_library(x)`. it is a similar problem I've had with test_util, because it is also a sublibrary which depends on the parent lib

I suggest to try a similar approach to test_util from PR https://github.com/erigontech/silkworm/pull/2575 by adding this to targets.cmake:

```
  if(NOT "${CMAKE_CURRENT_SOURCE_DIR}" MATCHES "/cmd$")
    list(FILTER SRC EXCLUDE REGEX "\/cmd\/")
  endif()
```

the effect would be to not include cmd files in the parent lib, and to include it in the cmd lib

this should work as long as the cmd is flat (no subfolders)
in the future we might improve both test_util and cmd cases by custom CMakeLists.txt search logic